ggshield/output/text/text_output_handler.py

```diff
@@ -32,6 +34,12 @@ def _process_scan_impl(self, scan: ScanCollection, top: bool = True) -> str:
 
         scan_buf.write(self.process_scan_results(scan))
 
+        current_result = scan_buf.getvalue()
+        if not current_result or current_result == secrets_engine_version():
```

Comment:
Checking the content of `current_result` sounds fragile. Since it's added a few lines above, maybe you can change the code to something like this?

```python
processed_scan_results = self.process_scan_results(scan)

scan_buf = StringIO()
if self.verbose:
    scan_buf.write(secrets_engine_version())
scan_buf.write(processed_scan_results)
if not processed_scan_results:
    scan_buf.write(
        no_new_leak_message() if scan.known_secrets_count else no_leak_message()
    )
```

Reply:
Looks cleaner indeed 👍  Thanks !

Reply:
you could also change to the condition 
```
if not scan.new_secrets_count:
    scan_buf.write(
        no_new_leak_message() if scan.known_secrets_count else no_leak_message()
    )
```